Assets/Code/Behaviours/GenerateCity.cs

```diff
@@ -94,6 +117,40 @@ void Start()
         // https://docs.unity3d.com/ScriptReference/Transform-parent.html?_ga=2.51063529.1903334404.1586621367-266296751.1584777261 
     }
 
+    public MapCell GetCellFromPosition(Vector3 position)
+    {
+        var relativePosition = position - bottomLeftCornerPosition;
+        var row = Mathf.FloorToInt(relativePosition.x) / Mathf.RoundToInt(cellSize);
+        var col = Mathf.FloorToInt(relativePosition.z) / Mathf.RoundToInt(cellSize);
+        try
+        {
+            return _map.GetMapCell(row, col);
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    public bool IsPositionOutOfBounds(Vector3 position)
+    {
+        return (position.x < bottomLeftCornerPosition.x || position.x > bottomLeftCornerPosition.x + Rows * cellSize ||
+                position.z < bottomLeftCornerPosition.z || position.z > bottomLeftCornerPosition.x + Cols * cellSize);
+    }
+
+    void Update()
+    {
+        foreach (var pedestrian in _pedestrians)
+        {
+            var positionOfLookingAhead = pedestrian.transform.position + pedestrian.Direction * _lookAheadDistanceMagnitude;
+            var cellOfLookingForward = IsPositionOutOfBounds(positionOfLookingAhead) ? null : GetCellFromPosition(positionOfLookingAhead);

```

Comment:
I have a suggestion about the update function for next iteration maybe.

behavior which is bound to the GenerateCity behavior to get the map (the same why i did with the behavior that puts the car randomly on the map). Than this behavior moves each pedestrian (As you already did) and can check the info on the map, the same way you did here :).